### DISCUSSION TITLE
Welcome to WGSExtract.github.io Discussions!

### DISCUSSION THREAD

**Discussion by RandyHarr** [WGSExtract/WGSExtract.github.io#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.
  
  Currently, Discussions are primarily hosted in a [Facebook group on Personal WGS testing](https://www.facebook.com/groups/373644229897409/).

**Comment by RobbieRRot** [comment]
Hello, I have came across a BAM header error while uploading my BAM file I've downloaded from European Nucleotide Archive. What happened and what should I do? The file definitely ends with .bam

**Reply by RandyHarr** [reply]
***UPDATE:  See the last comment in the thread here.  Appears some issues have cropped up with MacPorts and the WGS Extract v3 installer on MacOS Monterrey since the release around this time last year.  Workarounds are detailed there.

There can be many causes. Unrecognized reference model used to create the BAM.  A CRAM file and you have not downloaded the reference model first.  Improper compression format (not BGZIP'ed. etc.  If you can capture the command window behind the GUI menus, it may give us a clue to what went wrong.  Also, you can create a file (of any size and content) named .wgsedebug in your "home" directory.  This will turn on verbose debugging messages in that command window. We definitely could do better job at catching more errors and conditions and reporting them more explicitly.  If a file from the ENA, if you send me the link, I can download and try as well. If you know how to drop down into a command window, you can run the tool "htsfile" on your BAM file.  That will characterize key features like compression style, determined content, etc.

**Reply by RobbieRRot** [reply]
Thanks for the prompt reply! Here is the problem shown in the command window
![Screen Shot 2022-06-07 at 22 35 09](https://user-images.githubusercontent.com/107080010/172612423-48a3f38b-0d86-4180-9376-fe174dab4603.png)
As for the bam file I'm using: 
https://www.ebi.ac.uk/ena/browser/view/PRJEB36297?show=reads
this one 
![Screen Shot 2022-06-08 at 13 10 22](https://user-images.githubusercontent.com/107080010/172613016-36d33920-64d2-4abc-8243-27f97dafdec9.png)
quite near the end
Many thanks!

**Reply by RandyHarr** [reply]
The screen capture error indicates that the macports installation of samtools is not there.  Did your installation complete without error?  I would run the installer again and look at that command screen closely for any errors.  Macports puts its installed files in /opt/local.  

The Mac version of WGS Extract has an uninstall as well as install.  So you can safely run the uninstall to remove any Macports installation that may exist.  This may help get a new, clean installation.

I downloaded and processed the BAM just fine on my desktop.  Attached is the stats result from the alpha v4 release. Notice it is the standard hs37d5 reference model from the 1K genome project. Only 62 bp read length, single ended though.
![XW-M1R18 merged fixedHeader rmdup sorted q30_stats](https://user-images.githubusercontent.com/58348151/172686629-6168502b-1fa4-42a4-b00b-3b5db39ca142.jpg)
They appear to have added a M_ before the sequencer name which throws the software off with determining it was an Illumina HisSeq X sequencer with an 8 lane flow cell.  E00 is for HiSeq X with the CCXY being the 8 lane flow cell of that model. Here are the new Bin Stats showing they get about 25% coverage with 8 or more reads on the autosomes in this sample.
![XW-M1R18 merged fixedHeader rmdup sorted q30_binstats](https://user-images.githubusercontent.com/58348151/172686375-14643fa6-d5ca-4a2d-89e5-8793409516d3.jpg)

**Reply by RobbieRRot** [reply]
I tried reinstalling but the problem persists: here is the terminal from installation:

WGSExtract needs Python3, Macports, and Java to be installed on your computer.
It now checks if they and all the software they manage and we need is installed.
Should run this script with 'sudo -H' at the start. If not, will require a password now

Installing or updating Python3 ... (ignoring native Python3 in Catalina)
 ... Python3 already installed

Checking for needed Python3 modules ...
Password:
Requirement already satisfied: Pillow in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (9.1.1)
Requirement already satisfied: pyliftover in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (0.4)
Requirement already satisfied: pyscreenshot in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (3.0)
Requirement already satisfied: openpyxl in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (3.0.10)
Requirement already satisfied: pandas in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (1.4.2)
Requirement already satisfied: mss in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (6.1.0)
Requirement already satisfied: entrypoint2 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (1.0)
Requirement already satisfied: EasyProcess in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (1.1)
Requirement already satisfied: et-xmlfile in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from openpyxl) (1.1.0)
Requirement already satisfied: python-dateutil>=2.8.1 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (2.8.2)
Requirement already satisfied: numpy>=1.18.5 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (1.22.4)
Requirement already satisfied: pytz>=2020.1 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (2022.1)
Requirement already satisfied: six>=1.5 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

Checking Xcode CLI tools ...
... Xcode CLI tools OK
MacOSX Version: 10.14.6+build18G9323
Installing or updating MacPorts ...
  (*** WARNING: There are issues with Apple's M1 chip and MacPorts.           )
  (    See https://trac.macports.org/wiki/BigSurProblems for more information.)
 ... retrieving and installing MacPorts on MacOSX version 10.14.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
installer: Error - the package path specified was invalid: 'MacPorts-2.6.2-10.14-Mojave.pkg'.

Macports will now try to install or update packages (samtools, etc) if they are missing.
 ... MacPorts is retrieving needed packages and installing.
sudo: /opt/local/bin/port: command not found
/anaconda/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin
Installing Java ...
... nothing to install
Calling Upgrade script to finish WGS Extract install ...
*** OSX Specific Upgrades
sudo: /opt/local/bin/port: command not found
sudo: /opt/local/bin/port: command not found
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0     28      0 --:--:-- --:--:-- --:--:--    29
mv: rename bwa to /opt/local/bin/bwa: No such file or directory
Requirement already satisfied: pip in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (22.1.2)
Requirement already satisfied: Pillow in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (9.1.1)
Requirement already satisfied: pyliftover in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (0.4)
Requirement already satisfied: pyscreenshot in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (3.0)
Requirement already satisfied: openpyxl in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (3.0.10)
Requirement already satisfied: pandas in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (1.4.2)
Requirement already satisfied: psutil in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (5.9.1)
Requirement already satisfied: entrypoint2 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (1.0)
Requirement already satisfied: mss in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (6.1.0)
Requirement already satisfied: EasyProcess in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pyscreenshot) (1.1)
Requirement already satisfied: et-xmlfile in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from openpyxl) (1.1.0)
Requirement already satisfied: pytz>=2020.1 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (2022.1)
Requirement already satisfied: numpy>=1.18.5 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (1.22.4)
Requirement already satisfied: python-dateutil>=2.8.1 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from pandas) (2.8.2)
Requirement already satisfied: six>=1.5 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)
Installing WGS Extract v3 program and its templates ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  264M  100  264M    0     0  8126k      0  0:00:33  0:00:33 --:--:-- 10.7M
/Users/mac/Downloads/WGSExtractv3/Upgrade_UbuntuLinux.sh: line 211: 7z: command not found
cp: WGSExtractv3/*: No such file or directory
chmod: reference/genomes/*.sh: No such file or directory
finished installing WGS Extract v3 program and templates.
Press any key to close this window (first scroll up to review if desired) ...

Congratulations!  You finished installing WGS Extract v3!
 You can start WGS Extract v3 by clicking the Start_MacOSX.command file. For convenience, make an alias of it,
 rename to WGSExtractV3.command, and place it in your Application folder or on your desktop.
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

**Reply by RandyHarr** [reply]
*** UPDATE: Attached is a patched Install_MacOS.command file that changes the Macports curl command from "curl -O" to "curl -kO". Use this file to replace your current one. (Github required it be enclosed in a compressed file. Click to uncompress and then copy to replace your current one.)
[Install_MacOS.command.gz](https://github.com/WGSExtract/WGSExtract.github.io/files/8871348/Install_MacOS.command.gz)

Your full log of the install provides more specific information as to why Macports is not installing correctly.  I copy the section that is key here:
... retrieving and installing MacPorts on MacOSX version 10.14.
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
of Certificate Authority (CA) public keys (CA certs). If the default
bundle file isn't adequate, you can specify an alternate file
using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
the bundle, the certificate verification probably failed due to a
problem with the certificate (it might be expired, or the name might
not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
the -k (or --insecure) option.
installer: Error - the package path specified was invalid: 'MacPorts-2.6.2-10.14-Mojave.pkg'.

None of the MacOS installer "curl" commands have "-k" as that bypasses verifying it is the correct website and not been hacked.  If you wish to try and bypass this security check, you could add option "-k" to each "curl" command in the insstall_MacOS.command file.

Your Anaconda installation was likely not affecting any issue here and can likely be safely reinstalled.

***UPDATE: Seems this has been going on for awhile (certificate issues with MacOS and Macports sites; among others). https://trac.macports.org/ticket/51516 . Why it has not popped up before, I am not sure.  But seems MacOS uses a different trusted certificate authority than some other domains / URLs such that MacPorts certificates are not accepted by MacOS.  This should have occurred earlier but never has.  Not clear why.  The only work around is to turn off certificate checking under https: when doing the download (adding the -k option to the current curl -O command when downloading MacPorts).

***UPDATE: MacPorts has dropped support for MacOS 10.14 Mojave.  And thus moved their installer to a different, archived location.  We cannot support or guarantee this platform anymore.  We would encourage an upgrade to MacOS 10.15 Catalina; at minimum.  Support for Mojave has already been dropped in WGS Extract v4.

***UPDATE: It appears Numpy in the python packages now requires xcode to locally compile.  Even on the older Mojave you have.  We install xcode after Python as MacPorts needs it. This may cause an initial error and require the installer to be run twice on Mojave now.

***UPDATE: This v3 release from last year does fail on Monterrey as it was not yet released then and we did not know how they would number things and there was no Macports release for it back then.  I am checking the Mojave release that was supported on WGS Extract v3 (but has not been the last year on v4 in Alpha test).  Will take a bit because I have to recover that OS from backing store. We cannot easily update this old WGS Extract v3 release for Monterrey because the installer does not understand M1 arm64 versus Intel x86_64 architecture differences.

**Reply by RobbieRRot** [reply]
So what do I have to do now? Upgrade to Catalina or run the installer twice on Mojave?

**Reply by RandyHarr** [reply]
Long term, upgrade your OS as software needed to run WGS Extract is being archived on Mojave 10.14.  Ultimately, Monterrey 12 is best.

For now, just either edit and add one character to the Install_MacOS.command script (changing "curl -O" to "curl -kO" on the line referencing the MACPORTs install.  OR simply download the already edited file above, uncompress it, and replace your current Install_MacOS.command file with it.  Once you have this fix to the installer, you should be able to simply run the installer once; I believe.  

The mention of running the installer twice is an additional one that has come up but that did not seem to effect you.  (Likely because you already had run the Installer before the issue crept in with the Python libraries.)

**Reply by RobbieRRot** [reply]
I tried executing the edited file and came across this problem 
https://apple.stackexchange.com/questions/113974/full-privileges-on-command-file-but-it-says-i-dont-have-appropriate-privilege
that I don't know how to fix

**Reply by RandyHarr** [reply]
Sorry to hear you have continued issues.  

Back when you very first downloaded, you had to execute the .command files by ctrl-click right-click or and select "open".  Then you are given the option to "open" and bypass the security mechanism of running a command that was downloaded from the Internet.  Because this is a new, downloaded copy, the previous answer to approve this was removed.  See the manual installation section on MacOS for more details.

**Reply by RandyHarr** [reply]
I received an email of an additional response from you but do not see it posted here. You indicated the problem you are still having is that the .command did not uncompress with the execute permission set.  If you look in the user manual (https://bit.ly/35IziTY), it gives you the instructions to open a terminal window in the Installation directory.  Then you can type the command "chmod a+x *.command".  As far as I know, you cannot set the execute permission via the GUI / Window in MacOS.

**Reply by RobbieRRot** [reply]
Oh yes sorry about that, I just realized its in the manual and deleted the reply. I'm installing now.

**Reply by RobbieRRot** [reply]
Thank you so much it's working now!

**Reply by RandyHarr** [reply]
Glad to hear. Sorry you had to discover a change in behavior on Apple MacOS due to Apple making security certificate authority changes since the release.

**Comment by Uluxu** [comment]
Hi. I'm not sure how to explain what's happening, but how do I fix this? I know this has something to do with WGSE being unable to read the reference genome, but I'm unsure otherwise. 
Novice at bioinformatics!

Thank you so much
![vbgvbl](https://github.com/WGSExtract/WGSExtract.github.io/assets/158784161/1978194b-1370-44e1-b906-f491323868d9)

**Reply by RandyHarr** [reply]
What I see is an SNP definition file (All_SNPs_GRCh37_.ref_vcf.gz)  being used with a reference model (hg19_WGSE.fa.gz) where there is a mismatched naming convention.  GRCh uses the numeric naming convention (1, 2, 3, ...) whereas HG uses the historical one (chr1, chr2, ...).  The naming convention must match in the BAM and the two definition / reference files mentioned above. This mismatch of files should not occur.

What you are seeing is an error that it could not find sequence "1" in one of the files.

It is also strange that the hg19__WGSE reference model was chosen.  Did you have a situation where it could not figure out the reference model and asked you to choose?  Even if so, it should have only offered models with the same naming convention. Are your reference library files exactly as in the release?  Send the BAM header (see first tab) in text file form so I can see what is defined there for names in the BAM. 

The start of All_SNPs_GRCh37_.ref_vcf.gz showing the numeric naming is:
#CHROM  POS     ID
1       58814   rs114420996
1       69869   rs548049170
1       72526   rs547237130
1       74792   rs13328684

The hg19_WGSE.fa.gz reference model file header / dict showing the HG naming is (last few columns truncated for readability):
@HD	VN:1.0	SO:unsorted
@SQ	SN:chr10	LN:135534747	...
@SQ	SN:chr11	LN:135006516	...
@SQ	SN:chr12	LN:133851895	...

Hopefully the BAM content and header are not a mismatch.  That could possibly explain the two different name-convention files being chosen. Note the odd sort order of the hg19_WGSE model (chr10 first)  which does not match the ALL_SNPS one (1 chromosome first).  That is likely a problem as well.  I would also like to see the first few lines of the BAM to understand how it was actually sorted independent of what the BAM header indicates (the header is taken from the reference model DICT used to create the BAM). But lets simply start with the header.

The hg19_WGSE model is something put out with WGSE v1 that is grossly in error.  It was found at some esoteric ancient DNA site at a University and should never have been or be used.  We only still provide it in case any one used WGSE v1 and aligned to this model and created a CRAM.  You would need the model to realign to an unbroken reference model.  I guess I forgot the hg19_WGSE has the odd sort order and should negate it being used by the microarray generator inside the tool.  But still, I do not see how the tool selected those two different naming convention models.  Hence wanting to see your BAM header. 

In quick summary, you should realign the BAM to the standard HG19 or possibly HS37D5 reference model. And use that new BAM in the tool. But I would like to find out how the above happened as well.  If willing, I would maybe most easily diagnose by getting your BAM file itself.  If you can put it on a server somewhere and send me the URL link, that would be best.

**Reply by Uluxu** [reply]
Thank you very much for the quick response :) Sorry I hadn't seen this earlier.
It actually hadn't asked me to choose a reference genome, it came like that. Attached should be the header.
[Yana_old2.sort.rmdup.realign.md_header.txt](https://github.com/WGSExtract/WGSExtract.github.io/files/14158681/Yana_old2.sort.rmdup.realign.md_header.txt)
How would I go about realigning the bam to standard HG19 or HS37D5? Only by the red button in the tool?

I downloaded the bam from the European Nucleotide Archive here!
https://www.ebi.ac.uk/ena/browser/view/SAMEA5684443?show=reads

**Reply by RandyHarr** [reply]
So the BAM was done with an artificial reference model. We have seen this before.

It is Numeric (GRCh) naming. So the error is in the selection of the hg19_WGSE model which is traditional HG (chr1, etc) naming. The current WGS Extract does not have a numeric named, 25 SN, hg19 model that would match this BAM. 

Yes, the red realign button.  It is red simply because it takes some time.  But realize, many of these University samples have very poor coverage.  So even realigned, they may not generate a very useful microarray file.  They often get rejected by MyHeritage, GEDMatch etc as having the wrong proportion of homologous (same value in both copies of the autosome) vs heterozygous (each copy has a different value).

Release v5 (not yet available) has 3 of these esoteric, University submission reference models.  We had to deep scour the net to find them or even recreate them (with hopefully the same reference model values). The new tool will more readily handle them. With the caveat you should really realign them as soon as you notice such a reference model use.

Here are the stats from WGS Extract:
![image](https://github.com/WGSExtract/WGSExtract.github.io/assets/58348151/0c56d4e3-d719-4288-ab95-b80484213d73)
You really need more and longer reads to effectively get microarray results from the file. I think Miqui was eventually able to trim the microarray file to pass muster at GEDMatch for a similar sample / result from one of those studies.  I am suspecting these were sequenced 15+ years ago.  Both due to the size and age of the sequencer used. And the short read length (~50bp single end instead of 150 bp paired or effective 300 bp single end).

**Reply by Uluxu** [reply]
Very appreciated! Currently aligning the BAM file, supposed waiting time 9.0 hours but this is a big file so I suspect it'll be longer

I've actually used microarray on many BAM files from ancient individuals on ENA - it works very well most of the time. I've only encountered this error once before. Usually the only hindrance is that BAM files below ~50MB produce nothing or that some BAMs just have very low read depth, but it's extremely useful for GEDMATCH, YSEQ phenotype predictors, codegen.eu, so forth 
As most people cannot use Rsamtools by themselves at all

Looking forward to v5.0 for certain

In general, all Numeric models need to be realigned here as you said? I understand. BTW, do you have any other contacts?

**Reply by RandyHarr** [reply]
It was much less time due to the smaller number of reads. Only took 2 hrs on my machine which normally takes 8. 90gbase alignment is 160 CPU hrs. Each core is a CPU.

numeric models are fine. The 1k genome models (hsxxx) are numeric named. It is those models with only 24-25 SNs that have not been tuned for analysis that are the issue. As well as the low read count and often low quality reads (or damaged, frahmented DNA).

**Comment by DogCatSpace** [comment]
Hi Randy,

We are comparing with two genomes (myself and my daughter) from Nebula via WGSExtract.  Our genomes are so similar that it is statistically impossible unless: 1-We have the greatest incidence of uniparental disomy in history, 2-WGS is is not working as intended, or 3-Nebula is providing the same set of misreads for multiple genomes.  Is there any WGSE setting I could have mismanaged to produce this?  I know we ran different genomes because one is XX and the other XY.

**Reply by RandyHarr** [reply]
What do you mean exactly: "comparing genomes"? There is no feature in the tool to compare.

By chance, do you mean you extracted microarray files and are then comparing them? Which exact file (vendor, version)? Given you started with Nebula, I presume it did a liftover to build 37 as part of that. Is there any chance files with the same name got created and clobbered each other. Leading to the impression they are identical? How are you doing the compare?

**Comment by DogCatSpace** [comment]
Thanks for the reply.

I am manually comparing the text (from myself and my daughter) outputs from WGSExtract (WGSExtract-Dev46_13Jun2024).  For the "Select Microarray" we are using just "Everything, Combined file..."

These files are not even close to identical.  However, we are seeing what appears to be levels of uniparental disomy (UP) so high it is statistically impossible.  UP is where someone inherits two copies of a chromosome, or part of a chromosome, from one parent and none from the other.  If UP is not happening, the most likely explanation (of which I am aware) is that the chip used for the sequencing has a flaw for the genome portions where UP is apparent.

My question for you is whether it is possible this could also be problem related to WGSExtract?

**Comment by RandyHarr** [comment]
Anything is possible. You do realize that large portions of microarray files are the reference values, right? I would use tools like GEDMatch or DNA Kit Studio for the compare. 

Some forms of dysomy can be detected by microarray files. I would use the "are the parents related" tool in GEDMatch on the child file to help find indications of runs of homozygosity in the microarray file result.

**Comment by DogCatSpace** [comment]
Thanks.  Nope, stupidly I did not know that large portions of microarray files are the reference values, and that explains a lot of what I am seeing and sorry to have wasted your time.  Any suggested reading on this outside of the WGS Extract Manual V4?

**Comment by RandyHarr** [comment]
Maybe my glossary entry?
https://h600.org/wiki/Microarray+File+Formats

**Reply by DogCatSpace** [reply]
Thanks again Randy.

I am diving deeper into this to ensure I understand what is going on because others a running into the same problem and I can elaborate once I get clarification. Regarding "large portions of microarray files are reference values", am I correct in assuming these reference values came from the "male anonymous submitter"?  If so, this would explain why, regardless of what genome we look at, we are getting the same scary set of SNPs.  If that is the case, is it possible this anonymous submitter had these mutations?

All of this is stemming from a Reddit post that recommends using WGS Extract to generate Promethease files.

**Reply by RandyHarr** [reply]
I have no clue what reddit post you may be talking about. Regarding the reference model values, there was a combination of samples that created the initial reference models. The Y chromosome was primarily one submitter. But the models were normalized so rare variants are (for the most part) not in the model. Only the rCRS mitochondria model has not been so adjusted. Tools will usually adjust for VCFs if they were left in (and they can assume a reference value). Microarray files have all values and so it does not matter. They simply look for the ClinVar significant value; whether reference or not.

**Reply by DogCatSpace** [reply]
Regarding Reddit, see:

https://www.reddit.com/r/promethease/comments/1awuyqx/how_do_i_upload_nebula_wgs/
https://www.reddit.com/r/promethease/comments/1b24nkt/upload_from_nebula_to_promethease_brief_guide_and/

**Reply by DogCatSpace** [reply]
Regarding unwanted intrusion of the reference genome, we are seeing identical SNPs profiles from two distinct Nebula genomes, one male and the other female.  Two, of many, include:

X-linked adrenoleukodystrophy (5 SNPs)
Fanconi anemia (10 SNPs)

Is it possible this error is on our end or due to the reference genome?  This is not repeated when using other services, such as DNAgenics + Promethease.  Finally, there is no trace of any of these SNPs in our native Nebula files. 

![Untitled](https://github.com/WGSExtract/WGSExtract.github.io/assets/173186109/f87f14a4-6e0f-4769-a768-e04fafa9e498)

**Reply by RandyHarr** [reply]
A majority of those shown are no call (dash or '-'). How can that be bad?

**Reply by DogCatSpace** [reply]
Hi Randy,

May we safely assume these mutations are from the reference genome?

**Comment by DogCatSpace** [comment]
Super helpful site, thanks! so much for putting this all together.

**Comment by DogCatSpace** [comment]
Those terms “Bad” are from Promethease, not me.   And, Promethease has degraded significantly in the last few years, we are not supporters of this platform in any way.  

 

This does not address our primary question, “how do we get an identical set of mutations from two separate genomes from WGS?”.  Is this due to the reference genome?  Note there are MANY other calls that are not “no calls”.

**Comment by jotala** [comment]
Hello, I'm new to this thing
I did an Exome, but the issue is I got four files. (Lines?) 
I searched how to merge this files into one BAM with no success. 
I manager to install WGS on my Mac, I was able to Align the UBAN to BAM. But I still have split files.

My main goal its to covert to Microarray file formats. 


Bellow is what the Lab gives me.
Name on the website / downloaded filename
UBAN-1                                            /   XXX_S235_L001.unmapped.bam
UBAN-2                                           /   XXX_S235_L002.unmapped.bam
UBAN-3                                           /   XXX_S235_L003.unmapped.bam
UBAN-4                                           /   XXX_S235_L004.unmapped.bam
VCF_EXOMEDEPTH_GRCH38-1   /   XXX.cnv.GRCh38.vcf.gz
VCF_GATK_GRCH38-1                  /   XXX.variants.GRCh38.vcf.gz


Do you know how to combine multiples BAM / UBAM ?

**Reply by RandyHarr** [reply]
It appears you have an unmapped BAM for each lane in the sequencer flowcell. These could be single-end or paired-end reads. Best would be to create the FASTQ file(s) for each unmapped BAM. They can then be concatenated and used for alignment to create the BAM. It is simple but could be tricky / intricate. Might be easier to just do for you.

Note that a WES / Exome sequence may not have very good coverage of all the markets in a typical microarray file. My guess is your VCF files created from the BAMs would be your best source to use with tools. But many tools do not accept VCF so you may have to go this route.
